ReSharper.FSharp/src/FSharp.Psi.Features/src/CodeCleanup/FantomasHost.fs

```diff
@@ -2,15 +2,28 @@ namespace JetBrains.ReSharper.Plugins.FSharp.Services.Formatter
 
 open FSharp.Compiler.CodeAnalysis
 open FSharp.Compiler.Text
+open JetBrains.Core
 open JetBrains.Lifetimes
 open JetBrains.ProjectModel
 open JetBrains.Rd.Tasks
 open JetBrains.ReSharper.Plugins.FSharp.Fantomas.Client
 open JetBrains.ReSharper.Plugins.FSharp.Fantomas.Protocol
+open JetBrains.Util
+
+module internal Reflection =
+    let formatSettingType = typeof<FSharpFormatSettingsKey>
+
+    let getFieldValue obj fieldName =
+        let field = formatSettingType.GetField(fieldName)
+        if isNotNull field then field.GetValue(obj) else null
+
 
 [<SolutionComponent>]
 type FantomasHost(solution: ISolution, fantomasFactory: FantomasProcessFactory) =
     let mutable connection: FantomasConnection = null
+    let mutable formatConfigFields: string[] = [||]
+
+    let toEditorConfigName name = $"fsharp_{StringUtil.MakeUnderscoreCaseName(name)}"
```

Comment:
Maybe it would be better to reuse the `fsharp_` prefix literal with `FantomasSettingsConverter`. It's unlikely the prefix will change, but it would be easier to find its usages anyway.